python/feathub/processors/flink/table_builder/flink_table_builder.py

```diff
@@ -391,19 +393,36 @@ def _get_table_from_derived_feature_view(
             right_table_name,
             keys,
         ), right_table_join_field_descriptors in right_tables.items():
-            right_table = table_by_names[right_table_name].select(
-                *[
-                    native_flink_expr.col(right_table_field)
-                    for right_table_field in right_table_join_field_descriptors.keys()
-                ]
-            )
-            tmp_table = temporal_join(
-                self.t_env,
-                tmp_table,
-                right_table,
-                keys,
-                right_table_join_field_descriptors,
-            )
+            right_table = table_by_names[right_table_name]
+
+            if EVENT_TIME_ATTRIBUTE_NAME in right_table_join_field_descriptors.keys():
```

Comment:
Instead of relying on the existence of EVENT_TIME_ATTRIBUTE_NAME, would it be more readable and safer to do something like this:

```
if right_table is a series of DerivedFeatureView + RedisSource:
  do the new logic
  return

// Follow the existing logic. Throw exception if right_timestamp_field is None.
```